app/javascript/react/components/SessionFilters/MobileSessionFilters.tsx

```diff
@@ -79,24 +91,32 @@ const MobileSessionFilters = ({
     sessionType
   );
 
-  const fixedSessionTypeSelected: boolean = sessionType === SessionTypes.FIXED;
   const isIndoorParameterInUrl = isIndoor === "true";
   const airBeamSensorNameSelected = sensorName.startsWith("Air");
+  const isFixedSessionTypeSelected: boolean =
+    sessionType === SessionTypes.FIXED;
+  const isDormant = useAppSelector(selectIsDormantSessionsType);
 
   const sessionsCount = useMemo(() => {
     switch (sessionType) {
       case SessionTypes.FIXED:
         return isIndoorParameterInUrl
           ? indoorSessionsState.sessions.length
-          : fixedSessionsState.sessions.length;
+          : isDormant
+          ? dormantFixedSessionsState.length
+          : activeFixedSessionsState.length;
```

Comment:
I know it works well, but nested ternaries can get hard to read. It's okay to leave it as-is as it's not super long, but it could be good to try something more readable in case it grows.